src/types/helpers.ts

```diff
@@ -0,0 +1,36 @@
+import { None, Option } from '../option';
+import { Result, Err, Ok } from '../result';
+
+export type Primitives =
+  | number
+  | boolean
+  | string
+  | undefined
+  | null
+  | symbol
+  | bigint;
+
+// T means the current type in the recursion
+// L means the last type
+// O means the original type
+export type DeepFlattenContainers<T, L, O> = T extends Primitives
```

Comment:
Unfortunately I had a better formatting before apply the linter's `--fix` command